File: 13/13A.py
import sys

scanner_ranges = []
for i in range(100):
    scanner_ranges.append(0)

for line in sys.stdin:
    line = line.split(": ")
    scanner_ranges[int(line[0])] = int(line[1])

severity = 0
for time in range(100):
    if scanner_ranges[time] != 0 and time % (2*scanner_ranges[time]-2) == 0:
        severity += time * scanner_ranges[time]
print(severity)

File: 13/13A.cpp
#include <iostream>
#include <sstream>

using namespace std;

int main() {
    int scanner_ranges[100];
    for(int i = 0; i<100; ++i) {
        scanner_ranges[i] = 0;
    }
    string line;
    while(getline(cin, line)) {
        istringstream iss;
        cout << line << endl;
        string first, second;
        iss >> first >> second;
        cout << first << second << endl;
        // okay I don't know how to do this input in C++
        // rip
    }

    int total = 0;
    for(int time = 0; time < 100; ++time) {
        if(scanner_ranges[time] != 0 && time % scanner_ranges[time] != 0) {
            total++;
        }
    }
    cout << total << endl;
}
